cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -299,28 +300,49 @@ func (d *Daemon) mustInitializeStorage(cfg *config.Config) *feewindow.FeeWindows
 	defer cancelReadTxMeta()
 	var initialSeq uint32
 	var currentSeq uint32
-	dataMigrations, err := db.BuildMigrations(readTxMetaCtx, d.logger, d.db, cfg)
+	applicableRange, err := db.GetMigrationLedgerRange(readTxMetaCtx, d.db, cfg.HistoryRetentionWindow)
 	if err != nil {
-		d.logger.WithError(err).Fatal("could not build migrations")
+		d.logger.WithError(err).Fatal("could not get ledger range for migration")
 	}
 
 	// Merge migrations range and fee stats range to get the applicable range
 	latestLedger, err := db.NewLedgerEntryReader(d.db).GetLatestLedgerSequence(readTxMetaCtx)
-	if err != nil {
+	if err != nil && !errors.Is(err, db.ErrEmptyDB) {
 		d.logger.WithError(err).Fatal("failed to get latest ledger sequence: %w", err)
 	}
-
 	maxFeeRetentionWindow := max(cfg.ClassicFeeStatsLedgerRetentionWindow, cfg.SorobanFeeStatsLedgerRetentionWindow)
-	ledgerSeqRange := &db.LedgerSeqRange{FirstLedgerSeq: latestLedger - maxFeeRetentionWindow, LastLedgerSeq: latestLedger}
-	applicableRange := dataMigrations.ApplicableRange()
+	ledgerSeqRange := &db.LedgerSeqRange{FirstLedgerSeq: firstLedger, LastLedgerSeq: latestLedger}
+	if latestLedger > maxFeeRetentionWindow {
+		ledgerSeqRange.FirstLedgerSeq = latestLedger - maxFeeRetentionWindow
+	}
```

Comment:
it seems this ledger range calculation is equivalent to `db.GetMigrationLedgerRange()`